### DISCUSSION TITLE
contributions

### DISCUSSION THREAD

**Discussion by mateuszkornecki** [community/community#53850] [Repositories] [answered]
### Missing contributions 

Question

Hi! For some trasom my co tributions from [tree-frog](https://github.com/mateuszkornecki/tree-frog) repository is not counted at all. There is a chance that I did something wrong, I’m currently trying to rebuild my dev skills which I “lost” after a pretty serious accident - I almost died because of accident on my mountain bike and afterward I lost some of my memories. Sorry for this trivial question but I have no idea what I did wrong. Last activity was today,but GitHub is saying it was more than a week ago. Thanks and have a great day 😊 !

**Comment by airtower-luna** [comment, marked as answer]
GitHub assigns commits to users based on the *author email* recorded in the commit. The address in your commits in that directory [looks invalid](https://github.com/mateuszkornecki/tree-frog/commit/740f60803a4eb8d7d1b96fd829a8d09e76efb256.patch) (see the "From" line). Make sure your [user name and email are set correctly](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#_your_identity), for GitHub to recognize your commits `user.email` must be an address connected to your account. That'll only change *future* commits though.

For the existing commits you'll have to look at [rewriting history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History). Note that this will change *all* commit IDs after the removed (or changed) commit. This means anyone who already has a copy of your repository and tries to fetch updates will have to manually accept the modified history. :warning:

I've heard GitHub should recognize a [`.mailmap` file](https://git-scm.com/docs/gitmailmap) that maps the wrong email address to you as well, but didn't try that myself yet. If it works I'd love to hear about it, because that avoids rewriting history. :wink:

**Reply by mateuszkornecki** [reply]
Thanks a lot!! I just chaned it for futere commits, don't care about old one. Again thanks A LOT!

**Reply by airtower-luna** [reply]
Glad I could help! Please don't forget to mark the answer. :slightly_smiling_face:

**Comment by albertomorini** [comment]
Hello @airtower-luna , I have a kinda similar probelm.
I removed my old email from my account, then added the new one, my "contribnutions" are disappeared.
Then I re-added my old email to my profile, also make it primary, but the past contributions are still missing... Can you help me? Thanks a lot!
